src/main/java/io/vertx/java/spi/cluster/impl/jgroups/JGroupsClusterManager.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2011-2013 The original author or authors
+ * ------------------------------------------------------
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *     The Eclipse Public License is available at
+ *     http://www.eclipse.org/legal/epl-v10.html
+ *
+ *     The Apache License v2.0 is available at
+ *     http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.java.spi.cluster.impl.jgroups;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxException;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.impl.LoggerFactory;
+import io.vertx.core.shareddata.AsyncMap;
+import io.vertx.core.shareddata.Counter;
+import io.vertx.core.shareddata.Lock;
+import io.vertx.core.spi.cluster.AsyncMultiMap;
+import io.vertx.core.spi.cluster.ClusterManager;
+import io.vertx.core.spi.cluster.NodeListener;
+import io.vertx.java.spi.cluster.impl.jgroups.domain.ClusteredCounterImpl;
+import io.vertx.java.spi.cluster.impl.jgroups.domain.ClusteredLockImpl;
+import io.vertx.java.spi.cluster.impl.jgroups.listeners.TopologyListener;
+import io.vertx.java.spi.cluster.impl.jgroups.support.LambdaLogger;
+import org.jgroups.Channel;
+import org.jgroups.JChannel;
+import org.jgroups.blocks.atomic.CounterService;
+import org.jgroups.blocks.locking.LockService;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class JGroupsClusterManager implements ClusterManager, LambdaLogger {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JGroupsClusterManager.class);
+
+  public static final String JGROUPS_CONFIG_FILE = "vertx.jgroups.filename";
+  public static final String CLUSTER_NAME = "JGROUPS_CLUSTER";
+
+  private Vertx vertx;
+
+  private CacheManager cacheManager;
+
+  private JChannel channel;
+
+  private CounterService counterService;
+  private LockService lockService;
+
+  private String lock = "Lock";
+
+  private volatile boolean active = false;
+  private String address;
+  private TopologyListener topologyListener;
+
+  private final String jgroupsConfigurationFile;
+
+  public JGroupsClusterManager() {
+    this(System.getProperty(JGROUPS_CONFIG_FILE, "default-jgroups.xml"));
```

Comment:
The Vert.x convention is to load config from the classpath, i.e. look for a "jgroups.xml" and if not found use the default "default-jgroups.xml". Take a look at how it's done in HazelcastClusterManager.